src/Decoder.ts

```diff
@@ -650,7 +675,7 @@ export class Decoder<ContextType = undefined> {
   }
 
   private decodeString(byteLength: number, headerOffset: number): string | Uint8Array {
-    if (!this.useRawBinaryStrings || this.stateIsMapKey()) {
+    if (!this.useRawBinaryStrings || (!this.useMap && this.stateIsMapKey())) {
```

Comment:
If I read this right, does it mean if you are using maps and you set `useRawBinaryStrings` then it will use a uint8array for the map keys? I wonder if we should allow that to be switchable?

Reply:
That's correct, and your suggestion makes sense

Reply:
Added two options for this in 7342b02